### PR TITLE
Export send batch messages api in c style.

### PR DESCRIPTION
**Pull request by githublaohu**
## What is the purpose of the change
C batch sending interface

## Verifying this change
see example/CBatchProducer.c

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when a cross-module dependency exists.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: example/CBatchProducer.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  printf("Producer start.....\n");
+  StartSendMessage(producer);
+  ShutdownProducer(producer);
+  DestroyProducer(producer);
+  printf("Producer Shutdown!\n");
+}
+
+int main(int argc, char* argv[]) {
+  printf("Send Batch.....\n");
+  CreateProducerAndStartSendMessage();
```
Comment (ifplusor) [inline example/CBatchProducer.c]: only call CreateProducerAndStartSendMessage in main, so i think the function is unnecessary.

File: include/CBatchMessage.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#include "CMessage.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct CBatchMessage CBatchMessage;
+
+ROCKETMQCLIENT_API CBatchMessage* CreateBatchMessage();
+ROCKETMQCLIENT_API int addMessage(CBatchMessage* batchMsg, CMessage* msg);
```
[the remark this answers is not in the document]
Comment (githublaohu) [reply include/CBatchMessage.h]: Already processed

File: example/CBatchProducer.c (hunk 1 above)
Comment (githublaohu) [reply example/CBatchProducer.c]: The code structure is clear and consistent with the C case in other examples, which is easy to understand.

代码结构清晰点，同时与其他example中c案例一致，便于理解

File: src/extern/CBatchMessage.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#include <vector>
+
+#include "CBatchMessage.h"
+#include "CCommon.h"
+#include "CMessage.h"
+#include "MQMessage.h"
+
+using std::vector;
+
+#ifdef __cplusplus
```
[the remark this answers is not in the document]
Comment (githublaohu) [reply src/extern/CBatchMessage.cpp]: This is the C interface and must be ifdef
这是c接口，必须ifdef

[the remark this answers is not in the document]
Comment (ifplusor) [reply src/extern/CBatchMessage.cpp]: 这个肯定是要用c++编译器编译的，只要用了c++编译器，就一定会定制__cplusplus，所以你的说法是没道理的。c接口只要在声明的时候用extern “c”包起来就可以了，因为编译器生成的符号表有区别

[the remark this answers is not in the document]
Comment (ifplusor) [reply src/extern/CBatchMessage.cpp]: c接口应该只要在声明的时候用extern “c”包起来就可以了，.cpp实现的时候没必要再次中extern c。

[the remark this answers is not in the document]
Comment (githublaohu) [reply src/extern/CBatchMessage.cpp]: 其他extern实现类，都是这样实现的。如果需要进行整理，大家讨论下。

**Comment by jonnxu** [comment]
LGTM

**Pull request merged**